packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -217,14 +218,22 @@ class _DropdownMenuItemButtonState<T> extends State<_DropdownMenuItemButton<T>>
       child = Padding(padding: padding, child: child);
     }
     child = SizedBox(height: widget.route.itemHeight, child: child);
+
+    final bool isSelected = widget.itemIndex == widget.route.selectedIndex;
+    final FocusHighlightMode highlightMode = FocusManager.instance.highlightMode;
     // An [InkWell] is added to the item only if it is enabled
     if (dropdownMenuItem.enabled) {
       child = InkWell(
-        autofocus: widget.itemIndex == widget.route.selectedIndex,
+        autofocus: isSelected,
         enableFeedback: widget.enableFeedback,
         onTap: _handleOnTap,
         onFocusChange: _handleFocusChange,
-        child: child,
+        // On Desktop with highlightMode is traditional, the item is already drawn focus highlight whenever it is focused.
+        // However, on touch devices with highlightMode is touch, the item is not drawn focus highlight.
```

Comment:
Consider updating this comment.
Maybe:

```suggestion
        // When highlightMode is traditional, the InkWell draws the selected item background color.
        // When highlightMode is touch, insert an Ink to force the selected item backgound color. 
```